### PR TITLE
feat(js): Flatten localization prop type

### PR DESCRIPTION
**Pull request by desiprisg**
### What changed? Why was the change needed?

This PR flattens the localization prop, meaning that this:
```
localization={{
  inbox: {
    title: 'Inbox'
  }
}}
```

is now going to be specified as:
```
localization={{
  'inbox.title': 'Inbox'
}}
```
### Screenshots

Autocomplete is implemented as well:
Screenshot 2024-06-27 at 5 56 11 PM

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/src/ui/context/LocalizationContext.tsx [hunk 1]
```diff
@@ -21,12 +19,12 @@ const LocalizationContext = createContext<LocalizationContextType | undefined>(u
...
-  const localization = createMemo(() => deepMerge(defaultLocalization, props.localization || {}));
+  const localization = createMemo(() => ({ ...defaultLocalization, ...(props.localization || {}) }));
 
-  const t = (key: string) => {
-    const value = getNestedProperty(localization(), key);
+  const t = (key: LocalizationPath) => {
+    const value = localization()[key];
 
-    return (value || '') as string;
+    return value;
```
Comment (denis-kralj-novu) [inline packages/js/src/ui/context/LocalizationContext.tsx:25]: ❓ **question:** ‏ Is there a risk with the key not being present in the object and having the return value as `undefined`?

File: packages/js/src/ui/context/LocalizationContext.tsx [hunk 2]
```diff
@@ -21,12 +19,12 @@ const LocalizationContext = createContext<LocalizationContextType | undefined>(u
 type LocalizationProviderProps = ParentProps & { localization?: Localization };
 
 export const LocalizationProvider = (props: LocalizationProviderProps) => {
-  const localization = createMemo(() => deepMerge(defaultLocalization, props.localization || {}));
```
Comment (denis-kralj-novu) [inline packages/js/src/ui/context/LocalizationContext.tsx:24]: ❓ **question:** ‏ Is `deepMerge()` used anywhere else in the file? If not we can remove it.

Comment (denis-kralj-novu) [reply packages/js/src/ui/context/LocalizationContext.tsx:24]: same with `getNestedProperty()`

**Review by denis-kralj-novu — APPROVED** [review]
❓ **question:** ‏ I'm not really sure why this change is being done, is ther a place that gives more context for it?

**Comment by LetItRock** [comment]
> ❓ **question:** ‏ I'm not really sure why this change is being done, is ther a place that gives more context for it?

@denis-kralj-novu This PR is part of the new Inbox component. There is a guide that might give you a little more context: https://www.notion.so/novuhq/Inbox-Localization-76ebff747b7845a988d0f531f16f8a8d?pvs=4

**Review by LetItRock — APPROVED** [review]
awsome! 🙌

File: packages/js/src/ui/context/LocalizationContext.tsx (hunk 1 above)
Comment (desiprisg) [reply packages/js/src/ui/context/LocalizationContext.tsx:25]: There is always the `defaultLocalization` to fall back on, which will include all keys in `LocalizationPath`. I'm adding a `Required<>` as well to force us to include them all in there.

File: packages/js/src/ui/context/LocalizationContext.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/js/src/ui/context/LocalizationContext.tsx:24]: You're right!

**Pull request merged**